pysmt/substituter.py

```diff
@@ -20,11 +20,12 @@
 from six import iteritems
 
 import pysmt.walkers
+from pysmt.walkers.generic import handles
 import pysmt.operators as op
 from pysmt.exceptions import PysmtTypeError
 
 
-class Substituter(pysmt.walkers.DagWalker):
+class Substituter(pysmt.walkers.IdentityDagWalker):
```

Comment:
Right now (and possibly even in master) a user can instantiate a Substituter class and optains an identity walker. Can we prevent this by throwing a `NotImplementedError` in the walk?
What do you think?